javalib/src/main/scala/java/nio/ByteArrayBits.scala

```diff
@@ -1,9 +1,11 @@
 package java.nio
 
+import scala.scalanative.unsafe.Ptr
+
 // Ported from Scala.js
 private[nio] object ByteArrayBits {
   def apply(
-      array: Array[Byte],
+      array: Ptr[Byte],
```

Comment:
Probalby, in the place of each usage we already have Array[Byte] so this change probably can be reverted

Reply:
In HeapByteBuffer we use it to write. When we change ScalaArray to Ptr[Byte], changes to Ptr[Byte] are still saved to ScalaArray, but changing PtrArray to Array, the changes are not saved to the Ptr[Byte], as the data is copied from there, not cast. So we have to pass Ptr[Byte] there. As for concerns about the GC, ByteArrayBits is always used in companion of other classes, which hold the reference to the real Array, so the Ptr[] data will not be cleaned preemptively.